core/src/main/scala/neotypes/mappers.scala

```diff
@@ -278,8 +278,8 @@ object mappers {
         value match {
           case Some(v) =>
             Try(f(v)).toEither.left.map {
-              case ex: Uncoercible => IncoercibleException(ex.getLocalizedMessage, ex)
-              case ex: Throwable   => ex
+              case ex: Uncoercible => IncoercibleException(s"${ex.getLocalizedMessage} for field [$fieldName] with value [${v}]", ex)
```

Comment:
Just for the sake of style and standard, I would `for field [${fieldName}]` 
_(note the curly brackets)_

---

PS: This is a really minor thing and I wouldn't stop merging this only for this _(and simply fix it myself on my next pr)_.
However, the next comment does merit asking for a fix, so why not mentioning this too :p